framework/src/postprocessors/SideAdvectiveFluxIntegral.C

```diff
@@ -84,24 +84,30 @@ SideAdvectiveFluxIntegralTempl<is_ad>::computeFaceInfoIntegral(const FaceInfo *
 
   const auto state = determineState();
 
-  const auto adv_quant_face = raw_value((*_adv_quant)(
-      Moose::FaceArg({fi, Moose::FV::LimiterType::Upwind, true, false, nullptr}), state));
-
   // Get face value for velocity
-  const auto vel_x = raw_value(
-      (_vel_x)(Moose::FaceArg({fi, Moose::FV::LimiterType::Upwind, true, false, nullptr}), state));
+  const auto vel_x = raw_value((
+      _vel_x)(Moose::FaceArg({fi, Moose::FV::LimiterType::CentralDifference, true, false, nullptr}),
+              state));
   const auto vel_y =
-      _vel_y
-          ? raw_value((*_vel_y)(
-                Moose::FaceArg({fi, Moose::FV::LimiterType::Upwind, true, false, nullptr}), state))
-          : 0;
+      _vel_y ? raw_value((*_vel_y)(
+                   Moose::FaceArg(
+                       {fi, Moose::FV::LimiterType::CentralDifference, true, false, nullptr}),
+                   state))
+             : 0;
   const auto vel_z =
-      _vel_z
-          ? raw_value((*_vel_z)(
-                Moose::FaceArg({fi, Moose::FV::LimiterType::Upwind, true, false, nullptr}), state))
-          : 0;
+      _vel_z ? raw_value((*_vel_z)(
+                   Moose::FaceArg(
+                       {fi, Moose::FV::LimiterType::CentralDifference, true, false, nullptr}),
+                   state))
+             : 0;
+
+  const bool elem_is_upwind = RealVectorValue(vel_x, vel_y, vel_z) * _normals[0] >= 0;
+  const auto adv_quant_face = raw_value((*_adv_quant)(
+      Moose::FaceArg(
+          {fi, Moose::FV::LimiterType::CentralDifference, elem_is_upwind, false, nullptr}),
+      state));
 
-  return fi->normal() * adv_quant_face * RealVectorValue(vel_x, vel_y, vel_z);
+  return _normals[0] * adv_quant_face * RealVectorValue(vel_x, vel_y, vel_z);
```

Comment:
Was using `fi->normal()` not giving consistent results? It would be really nice not to use the qp-based `_normals` in the face info based computation

Reply:
Yes, I have an interior boundary set by SideSetsAroundSubdomainGenerator which set the normal direction of the sideset. I found the value gives by `fi->normal()` is not consistent with the preset normal of the sideset when distributed mesh is used.

Reply:
What about if you did this?
```c++
  return (_current_elem == fi->elemPtr() ? fi->normal() : Point(-fi->normal())) * adv_quant_face * RealVectorValue(vel_x, vel_y, vel_z);

Reply:
This works!